File: BidWinKoModify/app/src/main/java/com/bidwinko/screens/activity/TransactionActivity.java
package com.bidwinko.screens.activity;

import android.app.Activity;
import android.app.ProgressDialog;
import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.text.Html;
import android.util.Log;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;


import com.bidwinko.R;
import com.bidwinko.adapter.MYTransactionAdapter;
import com.bidwinko.API.APIService;

import com.bidwinko.model.Transaction;
import com.bidwinko.model.UserTransactionModel;
import com.bidwinko.utilies.Constants;
import com.bidwinko.API.Retrofit;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;


public class TransactionActivity extends AppCompatActivity {

    ArrayList<Transaction> myTransactionModelArrayList = null;
    MYTransactionAdapter myTransactionAdapter;
    RecyclerView recyclerView;

    ProgressDialog progressDialog;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_transaction);

        try{

            getSupportActionBar().setHomeButtonEnabled(true);
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            final Drawable upArrow = getResources().getDrawable(R.drawable.ic_arrow_back);
            upArrow.setColorFilter(getResources().getColor(R.color.white), PorterDuff.Mode.SRC_ATOP);
            getSupportActionBar().setHomeAsUpIndicator(upArrow);
            ((AppCompatActivity)this).getSupportActionBar().setTitle((Html.fromHtml("<font color=\"#ffffff\">" +getString(R.string.app_name)+" / "+getString(R.string.mytransaction) + "</font>")));


            recyclerView = (RecyclerView) findViewById(R.id.recycler_view);

//            getMyTransactions();

        }catch (Exception e){
            e.printStackTrace();
        }

    }

//    private void getMyTransactions() {
//
//        APIService service = Retrofit.getClient().create(APIService.class);
//        Call<UserTransactionModel> call = service.getUserTransactions(Constants.getSharedPreferenceInt(TransactionActivity.this,"userId",0),
//                        Constants.getSharedPreferenceString(TransactionActivity.this,"securitytoken",""),
//                        Constants.getSharedPreferenceString(TransactionActivity.this,"versionName",""),
//                        Constants.getSharedPreferenceInt(TransactionActivity.this,"versionCode",0),
//                        Constants.getSharedPreferenceString(TransactionActivity.this,"userFrom","") );
//
//
//
//        if(!((Activity) TransactionActivity.this).isFinishing()){
//            progressDialog = new ProgressDialog(TransactionActivity.this);
//            progressDialog.setMessage(getString(R.string.loadingwait));
//            progressDialog.show();
//            progressDialog.setCancelable(false);
//        }
//
//        call.enqueue(new Callback<UserTransactionModel>() {
//            @Override
//            public void onResponse(Call<UserTransactionModel> call, Response<UserTransactionModel> response) {
//                progressDialog.dismiss();
//                Log.e("testing", "onResponse: "+response.body().getMessage() );
//                try {
//                    if(response!=null) {
//                        if (response.isSuccessful()) {
//
//                            if (response.body().getStatus()==200) {
//                                myTransactionModelArrayList = response.body().getTransactions();
//
//                                myTransactionAdapter = new MYTransactionAdapter(myTransactionModelArrayList, TransactionActivity.this);
//                                RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(TransactionActivity.this);
//                                recyclerView.setLayoutManager(mLayoutManager);
//                                recyclerView.setItemAnimator(new DefaultItemAnimator());
//                                recyclerView.setAdapter(myTransactionAdapter);
//                            } else
//                                Toast.makeText(TransactionActivity.this, getString(R.string.systemmessage) + response.body().getMessage(), Toast.LENGTH_SHORT).show();
//                        } else
//                            Toast.makeText(TransactionActivity.this, getString(R.string.systemmessage) + response.body().getMessage(), Toast.LENGTH_SHORT).show();
//                    }
//                    else Toast.makeText(TransactionActivity.this, getString(R.string.systemmessage) + response.errorBody(), Toast.LENGTH_SHORT).show();
//                } catch (NullPointerException e) {
//                    Log.e("testing", "onFailure:exd "+e );
//                    Toast.makeText(TransactionActivity.this, getString(R.string.systemmessage)+e, Toast.LENGTH_SHORT).show();
//
//                }
//            }
//
//            @Override
//            public void onFailure(Call<UserTransactionModel> call, Throwable t) {
//                progressDialog.dismiss();
//                try {
//                    Toast.makeText(TransactionActivity.this, getString(R.string.systemmessage)+t.getMessage(), Toast.LENGTH_SHORT).show();
//
//                }catch (NullPointerException e) {
//
//                    Toast.makeText(TransactionActivity.this, getString(R.string.systemmessage)+e, Toast.LENGTH_SHORT).show();
//                }
//            }
//        });
//    }


    @Override
    public void onBackPressed() {
        super.onBackPressed();
        this.finish();
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    @Override
    protected void onPause() {
        dismissProgressDialog();
        super.onPause();


    }

    @Override
    public void onStop() {
        dismissProgressDialog();
        super.onStop();


    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    private void dismissProgressDialog() {
        if (progressDialog != null && progressDialog.isShowing()) {
            progressDialog.dismiss();
        }
    }

}

File: BidWinKoModify/app/src/main/java/com/bidwinko/screens/activity/TransactionActivity.kt
package com.bidwinko.screens.activity

import android.animation.Animator
import android.app.ProgressDialog
import android.graphics.PorterDuff
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.text.Html
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import com.bidwinko.R
import com.bidwinko.adapter.MYTransactionAdapter
import com.bidwinko.databinding.ActivityTransactionBinding
import com.bidwinko.model.RequestModels.CommonRequest
import com.bidwinko.model.ResponseModels.Transaction
import com.bidwinko.utilies.Constants
import com.bidwinko.utilies.SessionManager
import com.bidwinko.viewModel.mainViewModel

class TransactionActivity : AppCompatActivity() {
    var myTransactionModelArrayList: ArrayList<Transaction> = ArrayList()
    var myTransactionAdapter: MYTransactionAdapter? = null
    var recyclerView: RecyclerView? = null
    var progressDialog: ProgressDialog? = null
    lateinit var binding: ActivityTransactionBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityTransactionBinding.inflate(layoutInflater)
        setContentView(binding.root)
        try {
            supportActionBar!!.setHomeButtonEnabled(true)
            supportActionBar!!.setDisplayHomeAsUpEnabled(true)
            supportActionBar!!.setBackgroundDrawable(ColorDrawable(resources.getColor(R.color.black)))
            val upArrow = getResources().getDrawable(R.drawable.ic_arrow_back)
            upArrow.setColorFilter(getResources().getColor(R.color.white), PorterDuff.Mode.SRC_ATOP)
            supportActionBar!!.setHomeAsUpIndicator(upArrow)
            (this as AppCompatActivity).supportActionBar!!.title = Html.fromHtml(
                "<font color=\"#ffffff\">" + getString(R.string.mytransaction) + "</font>"
            )
            recyclerView = findViewById<View>(R.id.bid_recycler_view) as RecyclerView
            getMyTransactions()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    private fun getMyTransactions() {
        if (!(this.isFinishing)) {
            progressDialog = ProgressDialog(this)
            progressDialog!!.setMessage(getString(R.string.loadingwait))
            progressDialog!!.show()
            progressDialog!!.setCancelable(false)
        }

        val homeRequest = CommonRequest(
            userId = SessionManager(this).GetValue(Constants.USER_ID).toString(),
            securityToken = SessionManager(this).GetValue(Constants.SECURITY_TOKEN),
            versionName = SessionManager(this).GetValue(Constants.VERSION_NAME),
            versionCode = SessionManager(this).GetValue(Constants.VERSION_CODE)
        )

        mainViewModel(this).GetTransactionHistory(homeRequest).observe(this)
        {
            dismissProgressDialog()
            if (it.status == 200) {
                if (!it.transactions.isNullOrEmpty() && it.transactions.size > 0) {
                    myTransactionModelArrayList.addAll(it.transactions)
                    recyclerView?.adapter = MYTransactionAdapter(myTransactionModelArrayList, this)
                } else {
                    binding.bidRecyclerView.visibility = View.GONE
                    binding.nodatafound.visibility = View.VISIBLE
                    binding.animm.addAnimatorListener(object : Animator.AnimatorListener {
                        override fun onAnimationStart(animation: Animator) {
                        }

                        override fun onAnimationEnd(animation: Animator) {
                            binding.animm.playAnimation()
                        }

                        override fun onAnimationCancel(animation: Animator) {
                        }

                        override fun onAnimationRepeat(animation: Animator) {
                        }

                    })
                }
            } else {
                binding.wrong.visibility = View.VISIBLE
                binding.anim.playAnimation()
                binding.anim.let {
                    it.addAnimatorListener(object : Animator.AnimatorListener {
                        override fun onAnimationStart(animation: Animator) {
                        }

                        override fun onAnimationEnd(animation: Animator) {
                            it.playAnimation()
                        }

                        override fun onAnimationCancel(animation: Animator) {
                        }

                        override fun onAnimationRepeat(animation: Animator) {
                        }

                    })
                }
                binding.bidRecyclerView.visibility = View.GONE
            }
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    override fun onPause() {
        dismissProgressDialog()
        super.onPause()
    }

    public override fun onStop() {
        dismissProgressDialog()
        super.onStop()
    }

    private fun dismissProgressDialog() {
        if (progressDialog != null && progressDialog!!.isShowing) {
            progressDialog!!.dismiss()
        }
    }
}